packages/medusa/src/api-v2/store/customers/me/addresses/route.ts

```diff
@@ -29,7 +30,7 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
 }
 
 export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
-  const customerId = req.auth_user!.app_metadata.customer_id
+  const customerId = req.auth!.actor_id!
```

Comment:
Let's check that there indeed is an actor ID, instead of forcing the typings.

Reply:
I think the problem here is the lack of typings on the MedusaRequest, since this route is guarded by the auth middleware, I suppose the auth.actor_id will be present. When we will change the MedusaRequest to accept some template arugment we should be able to get rid of those wdyt?

Reply:
But if you forget to apply the middleware I guess it could be undefined, so as you suggested I guess it would be nice to have 2 types, Unauthenticated and Authenticated, which will solve this issue. But I'm fine with it as it is for now 👍

Reply:
fine too, it is more to improve the DX, but even though the type says it is present it might not be if the middleware is not applied. I ve suggested the split elsewhere and it can serve as reference